### ISSUE TITLE
Template project errors

### ISSUE THREAD

**Issue by JasperCiti** [unoplatform/uno#7532]
### Current behavior

1. I had run "dotnet tool install -g Uno.Check --version 0.2.0-dev.327", it installed a lot of SDK's and stuff.
2. Then I created a "Multi-Platform App (Uno Platform|net6)" project. 
3. I switched to App1.Mobile

When build I get these errors:
Specified AndroidManifest file does not exist: C:\Users\xxx\source\repos\App1\App1\App1.Mobile\AndroidManifest.xml.

Which is valid the AndroidManifest.xml is in a different folder: C:\Users\xxx\source\repos\App1\App1\App1.Mobile\ **Android** \AndroidManifest.xml

Another error:
Platform version is not present for one or more target frameworks, even though they have specified a platform: net6.0-ios, net6.0-maccatalyst

Granted. I'm running Windows. How do I get rid of this?

Another error:
To build this project, the following workloads must be installed: android
To install these workloads, run the following command: dotnet workload install android

This doesn't make sense because it was already installed by the above mentioned Uno.Check tool.

### Expected behavior

The template project should work out-of-the-box without manual project file hacking.

### Affected platforms

Android

### IDE

Visual Studio 2022

### IDE version

17.0.1

**Comment by mikernet** [comment]
If you use the dev branch stuff right now you're going to have issues. They are in the process of major changes across the board.

**Comment by jeromelaban** [comment]
Try running uno.check without the version parameter:
```
dotnet tool update -g Uno.Check
```

**Comment by brinawebb** [comment]
I have the exact same problem in the Uno Mobile project template. This is from the newly published uno project templates after the conference today.

VS2022 v 17.0.2
Brand new Windows 11 machine, almost nothing installed except VS2022
Mobile Workload is installed in Visual Studio (I uninstalled and reinstalled it)
uno-check is successful.
I can successfully run the UWP and Wasm templates.

![image](https://user-images.githubusercontent.com/4342718/144153828-64436921-7d86-4065-87ae-d62dc44a6f6a.png)

There may be a missing version in one of your dependencies? When I run "dotnet workload install ios" I get the below error. Going to nuget for Microsoft.iOS.Sdk there is no version 15.0.101-preview.11.391

![image](https://user-images.githubusercontent.com/4342718/144154384-2762ecdc-9955-4cd5-bb8b-8af23fc3f2cf.png)

**Comment by jeromelaban** [comment]
Thanks for the details. Installing the workloads manually from the command-line is causing these issues, as uno-check install explicit versions.

Can you post the binlog of your app's build ? See https://platform.uno/docs/articles/uno-builds-troubleshooting.html#generating-msbuild-binary-log-files

**Comment by brinawebb** [comment]
OK, thanks for your response! I could only use the commandline from your link there as the log viewer and project system tool don't support VS2022. I've attached two files though, one from msbuild at the command prompt and the other from the VS build output window.

[msbuildlog.txt](https://github.com/unoplatform/uno/files/7630060/msbuildlog.txt)

[vsbuildoutput.txt](https://github.com/unoplatform/uno/files/7630073/vsbuildoutput.txt)

**Comment by brinawebb** [comment]
Also, not sure if this helps you at all (probably better to have more info than too little). When I go to the nuget package for iOS it only lists version 15.0.101-preview.10.251
https://www.nuget.org/packages/Microsoft.iOS.Windows.Sdk/

The workload failure from above is indicating that it cannot find 15.0.101-preview.11.391

Am I maybe missing a package source?

**Comment by jeromelaban** [comment]
The msbuild logging tools are [supported in VS 2022](https://marketplace.visualstudio.com/items?itemName=VisualStudioProductTeam.ProjectSystemTools2022) (We'll update the docs). Can you provide the logs using these, thanks ?

**Comment by JasperCiti** [comment]
> Try running uno.check without the version parameter:
> 
> ```
> dotnet tool update -g Uno.Check
> ```

Produces

```
C:\Program Files\dotnet\sdk\6.0.100\NuGet.targets(130,5): warning : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.
C:\Program Files\dotnet\sdk\6.0.100\NuGet.targets(130,5): warning : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.
C:\Program Files\dotnet\sdk\6.0.100\NuGet.targets(130,5): warning : Your request could not be authenticated by the GitHub Packages service. Please ensure your access token is valid and has the appropriate scopes configured.
C:\Program Files\dotnet\sdk\6.0.100\NuGet.targets(130,5): error : Failed to retrieve information about 'uno.check' from remote source 'https://nuget.pkg.github.com/RevenantX/download/uno.check/index.json'.
C:\Program Files\dotnet\sdk\6.0.100\NuGet.targets(130,5): error :   Response status code does not indicate success: 401 (Unauthorized).
Tool 'uno.check' failed to update due to the following:
The tool package could not be restored.
Tool 'uno.check' failed to install. This failure may have been caused by:

* You are attempting to install a preview release and did not use the --version option to specify the version.
* A package by this name was found, but it was not a .NET tool.
* The required NuGet feed cannot be accessed, perhaps because of an Internet connection problem.
* You mistyped the name of the tool.

For more reasons, including package naming enforcement, visit https://aka.ms/failure-installing-tool
```

**Comment by brinawebb** [comment]
https://github.com/unoplatform/uno/issues/7532#issuecomment-985015810
Hello Jerome. Here is the build log. To create this I made a new project from the solution templates (removed all of the skia projects) and built the desktop fine, then the Mobile project. The binlog is from the build of the mobile project. Note that this was created with a new Windows 11 machine that only has VS 2022.
[_Debug_AnyCPU_Build_2021-12-05T06_19_00.4454644-08_00.zip](https://github.com/unoplatform/uno/files/7656052/_Debug_AnyCPU_Build_2021-12-05T06_19_00.4454644-08_00.zip)

I also updated your project templates and when creating a solution using the commandline I get similar errors (attached)
[dotnetnewoutput.txt](https://github.com/unoplatform/uno/files/7656068/dotnetnewoutput.txt)

**Comment by brinawebb** [comment]
I figured out my issue... If someone else runs into this. Not sure if this will fix the issue for the OP, but somehow I had the .NET6 SDK 6.0.101 installed? I uninstalled it, reinstalled 6.0.100, then ran uno-check, which then installed all of the missing workloads. All up and going now.

**Comment by JasperCiti** [comment]
No, I have Microsoft .NET SDK 6.0.100 (x64) installed.

**Comment by jeromelaban** [comment]
@JasperCiti your error comes from the fact that you have a invalid (or logged-in) source in your nuget.config. You can use the `--ignore-failed-sources` parameter to `dotnet tool install` to get the tool to skip it.

**Comment by joer33304** [comment]
This also fails with a new install and latest vs 2022 which installs SDK 6.0.2. The check tries to install 6.0.1 which I think is the issue. But there are also other weird errors like "package no signed". See log output:

» Attempting to fix: .NET SDK - Workloads (6.0.100)
SHELL: C:\Program Files\dotnet\dotnet.exe workload update --no-cache --disable-parallel --from-rollback-file "C:\Users\Joe\AppData\Local\Temp\maui-check-25d57714\workload.json" --source "https://pkgs.dev.azure.com/xamarin/public/_packaging/maui-main/nuget/v3/index.json" --source "https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json" --source "https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json" --source "https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet6/nuget/v3/index.json" --source "https://api.nuget.org/v3/index.json"

No workloads installed for this feature band. To update workloads installed with earlier SDK versions, include the --from-previous-sdk option.
Updated advertising manifest microsoft.net.sdk.android.
Updated advertising manifest microsoft.net.sdk.ios.
Updated advertising manifest microsoft.net.sdk.maccatalyst.
Updated advertising manifest microsoft.net.sdk.macos.
Updated advertising manifest microsoft.net.sdk.maui.
Updated advertising manifest microsoft.net.sdk.tvos.
Updated advertising manifest microsoft.net.workload.emscripten.
Failed to update the advertising manifest microsoft.net.workload.mono.toolchain: Failed to validate package signing.

Verifying Microsoft.NET.Workload.Mono.ToolChain.Manifest-6.0.200.6.0.2-mauipre.1.22102.15

error: NU3004: The package is not signed.

Package signature validation failed.

.
Downloading microsoft.net.sdk.android.manifest-6.0.200.msi.x64 (31.0.101-preview.11.81)
Workload installation failed. Rolling back installed packs...
Workload update failed: One or more errors occurred. (microsoft.net.sdk.android.manifest-6.0.200.msi.x64::31.0.101-preview.11.81 is not found in NuGet feeds https://pkgs.dev.azure.com/xamarin/public/_packaging/maui-main/nuget/v3/index.json;https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json;https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json;https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet6/nuget/v3/index.json;https://api.nuget.org/v3/index.json".)
Fix failed - Workload Update failed: `dotnet workload update --no-cache --disable-parallel --from-rollback-file
"C:\Users\Joe\AppData\Local\Temp\maui-check-25d57714\workload.json" --source
"https://pkgs.dev.azure.com/xamarin/public/_packaging/maui-main/nuget/v3/index.json" --source
"https://pkgs.dev.azure.com/azure-public/vside/_packaging/xamarin-impl/nuget/v3/index.json" --source
"https://pkgs.dev.azure.com/dnceng/public/_packaging/dotnet6/nuget/v3/index.json" --source
"https://dnceng.pkgs.visualstudio.com/public/_packaging/dotnet6/nuget/v3/index.json" --source "https://api.nuget.org/v3/index.json"`
──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────

**Comment by jeromelaban** [comment]
Thanks VS 2022 17.2 Preview 1 support will be available soon in Uno 4.1, but you can take a look at this discussion for installing it properly: https://github.com/unoplatform/uno/discussions/7833

**Comment by jeromelaban** [comment]
Uno 4.1 is now available. Make sure to review the content of https://github.com/unoplatform/uno/discussions/7833.
